snappy_test.go

```diff
@@ -322,6 +323,18 @@ func downloadTestdata(b *testing.B, basename string) (errRet error) {
 			os.Remove(filename)
 		}
 	}()
+	// Generate random data
+	if basename == "random" {
+		rng := rand.New(rand.NewSource(27354294))
+		for i := 0; i < 1<<20; i++ {
+			_, err := f.Write([]byte{byte(rng.Intn(255))})
```

Comment:
Ugh, can you write bigger batches? One million syscalls, each doing a tiny one-byte write, seems relatively expensive.

Also, s/255/256/